operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,309 @@
+            throw new Error("There has been a parsing error in the operation definition of the $everything operation.", e);
+        } catch (IOException e) {
+            throw new Error("There has been an IO error reading the operation definition of the $everything operation.", e);
+        }
+    }
+
+    @Override
+    protected Parameters doInvoke(FHIROperationContext operationContext, Class<? extends Resource> resourceType, String logicalId, String versionId, Parameters parameters, FHIRResourceHelpers resourceHelper) throws FHIROperationException {
+        LOG.entering(this.getClass().getName(), "doInvoke");
+
+        Patient patient = null;
+        try {
+            patient = (Patient) resourceHelper.doRead(PATIENT, logicalId, false, false, null, null);
+        } catch (Exception e) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("An unexpected error occurred while reading patient " + logicalId, IssueType.EXCEPTION);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+        if (patient == null) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("Patient with ID " + logicalId + " does not exist.", IssueType.NOT_FOUND);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+
+        Entry patientEntry = buildPatientEntry(operationContext, patient);
+        List<Entry> allEntries = new ArrayList<>(SearchConstants.MAX_PAGE_SIZE);
+        allEntries.add(patientEntry);
+        
+        // We can't always use the "date" query parameter to query by clinical date, only with some resources.
+        // Initial list obtained from the github issue: https://github.com/IBM/FHIR/issues/1044#issuecomment-769788097
+        // Otherwise the search throws an exception. We create a params map with and without and use as needed
+        MultivaluedMap<String, String> queryParameters = parseQueryParameters(parameters);
+        MultivaluedMap<String, String> queryParametersWithoutDates = new MultivaluedHashMap<String, String>(queryParameters);
+        queryParametersWithoutDates.remove(DATE_QUERY_PARAMETER);
+        
+        List<String> resourceTypesOverride = getOverridenIncludedResourceTypes(parameters);
+        List<String> resourceTypes = resourceTypesOverride.isEmpty() ? defaultResourceTypes : resourceTypesOverride;
+        
+        for (String compartmentType : resourceTypes) {
+            MultivaluedMap<String, String> searchParameters = queryParameters;
+            if (!SUPPORT_CLINICAL_DATE_QUERY.contains(compartmentType)) {
+                searchParameters = queryParametersWithoutDates;
+            }
+            Bundle results = null;
+            try {
+                results = resourceHelper.doSearch(compartmentType, PATIENT, logicalId, searchParameters, null, null, null);
```

Comment:
If you're getting ALL results for a resource type, will have to call doSearch in a loop. Since there's a max page size the persistence layer will return, results might come back paged.
If there's a hard limit that you won't exceed, need to check here and throw an exception if exceeded.

Reply:
I am not sure I completely follow, but I am not getting "all" resources of a type only the first 1000. 
This initial implementation was based on https://github.com/IBM/FHIR/issues/1044#issuecomment-769788097 saying that paging was not supported initially.

Reply:
Done.